src/Workspaces/CSharp/Portable/Formatting/Rules/TokenBasedFormattingRule.cs

```diff
@@ -153,7 +153,7 @@ public override AdjustNewLinesOperation GetAdjustNewLinesOperation(SyntaxToken p
                     attributeOwner is MemberDeclarationSyntax ||
```

Comment:
now this special condition no longer needed since default behavior is basically what you want. I would just delete whole code (from line 150 to 157)

Reply:
Yes you are right. I kept the change as simple as possible because this also introduces a change in behavior.

**Example 1**

Code
```CSharp
[Attr1] [Attr2]  
[Attr3] [Attr4]
int field;
```

Previously
```CSharp
[Attr1]
[Attr2]  
[Attr3]
[Attr4]
int field;
```

Now
```CSharp
[Attr1] [Attr2]  
[Attr3] [Attr4]
int field;
```

**Example 2**

Code
```CSharp
[Attr1] [Attr2]  
[Attr3] [Attr4] int field;
```

Previously
```CSharp
[Attr1]
[Attr2]  
[Attr3] [Attr4] int field;
```

Now
```CSharp
[Attr1] [Attr2]  
[Attr3] [Attr4] int field;
```

This might surprise people that enter code like this:

```CSharp
[Attr1] [Attr2] [Attr3] [Attr4] 
int field;
```

and expect it to be formatted like this

```CSharp
[Attr1]
[Attr2]
[Attr3]
[Attr4] 
int field;
```

You should discuss internally what the desired behavior should look like.

Reply:
> You should discuss internally what the desired behavior should look like.

In each of your examples, the "previously" code is correct.

Reply:
There are three options to achieve that:

1. Revert the last commit and keep the nested suppressions.
2. Revert the last commit and enhance the suppressions logic to avoid overlapping suppressions (break the loop). This requires the detection whether two tokens are on the same line.
3. Further enhance the logic in TokenBasedFormattingRule. This also requires the detection whether two tokens are on the same line.

I personally would preffer 1.

Reply:
:memo: I replied where you added those earlier comments